src/sc/sidechaintypes.cpp

```diff
@@ -304,6 +304,14 @@ wrappedScProofPtr CScProof::GetProofPtr() const
     std::lock_guard<std::mutex> lk(_mutex);
     if (proofData == nullptr)
     {
+        if (byteVector.size() > MaxByteSize())
```

Comment:
As I saw, we have an assert in CScProof constructor and `SetByteArray` method:
`assert(byteArrayIn.size() <= this->MaxByteSize());`
Do we need additional check here?